clash-lib/src/Clash/Normalize/PrimitiveReductions.hs

```diff
@@ -73,6 +79,172 @@ import           Clash.Rewrite.Types
 import           Clash.Rewrite.Util
 import           Clash.Unique
 import           Clash.Util
+import qualified Clash.Util.Interpolate           as I
+
+typeNatAdd :: TyConName
+typeNatAdd =
+  Name User "GHC.TypeNats.+" (getKey typeNatAddTyFamNameKey) wiredInSrcSpan
+
+typeNatMul :: TyConName
+typeNatMul =
+  Name User "GHC.TypeNats.*" (getKey typeNatMulTyFamNameKey) wiredInSrcSpan
+
+typeNatSub :: TyConName
+typeNatSub =
+  Name User "GHC.TypeNats.-" (getKey typeNatSubTyFamNameKey) wiredInSrcSpan
+
+vecHeadPrim
+  :: TyConName
+  -- ^ Vec TyCon name
+  -> Term
+vecHeadPrim vecTcNm =
+  Prim (PrimInfo "Clash.Sized.Vector.head" (vecHeadTy vecTcNm) WorkNever)
+
+vecLastPrim
+  :: TyConName
+  -- ^ Vec TyCon name
+  -> Term
+vecLastPrim vecTcNm =
+  Prim (PrimInfo "Clash.Sized.Vector.last" (vecHeadTy vecTcNm) WorkNever)
+
+vecHeadTy
```

Comment:
Comment with the type signature above this so it's more obvious that this is correct for `vecLastPrim`. There's a slight dissonance seeing `primA` have type `primBType`